docs/design/sql/05-ddl.md

```diff
@@ -0,0 +1,305 @@
+# SQL DDL
+
+## Overview
+
+This part contains reference information for the DDL commands supported by Hazelcast Mustang.
+
+## Table of Contents
+1. [CREATE EXTERNAL TABLE](#1-create-external-table) -- define a new external table
+2. [CREATE EXTERNAL TABLE AS](#2-create-external-table-as) -- define a new external table from the results of a query
+3. [DROP EXTERNAL TABLE](#3-drop-external-table) -- remove external table
+
+## 1 CREATE EXTERNAL TABLE
+
+`CREATE EXTERNAL TABLE` -- define a new external table
+
+### 1.1 Synopsis
+
+```sql
+CREATE [OR REPLACE] EXTERNAL TABLE [IF NOT EXISTS] table_name [ (
+ column_name column_type [ EXTERNAL NAME column_external_name ] [, ... ]
+) ]
+TYPE table_type
+[ OPTIONS ( table_parameter table_parameter_value [, ... ] ) ]
+```
+
+### 1.2 Description
+
+`CREATE EXTERNAL TABLE` registers a virtual table that references an external storage system. For some storage systems,
+it does not create a physical entity until a write occurs. Column list is optional, as for some storage systems and
+serialization formats, it is possible to infer it.
+
+The table is created in the `public` schema. The name of the table must be distinct from the name of any other table in
+the schema.
+
+### 1.3 Parameters
+
+**OR REPLACE**
+
+Replace table definition if one with the same name already exists.
+
+**IF NOT EXISTS**
+
+Do not fail if a table with the same name already exists.
+
+**table_name**
+
+The unqualified name of the table to be created.
+
+**column_name**
+
+The name of the column in the table.
+
+**column_type**
+
+The data type of the column. For more information on the supported data types, refer to [type system document](https://github.com/hazelcast/hazelcast/blob/master/docs/design/sql/01-type-system.md).
+
+**column_external_name**
+
+The path to the field in the external storage system. Used to link a column to physical field under a different name
+(i.e. to be able to access both key and value `IMap` fields having same name).
+
+**table_type**
+
+The external storage system identifier. Supported table types are listed in Table 1. Not all storage systems implement
+full set of operations, so for instance you might be able to `SELECT` from a given table but not `INSERT` into it.
+
+*Table 1: CREATE EXTERNAL TABLE supported table types*
+
+| Table type | Description |
+|---|---|
+| com.hazelcast.IMap | Table backed by Hazelcast `IMap` |
+| com.hazelcast.ReplicatedMap | Table backed by Hazelcast `ReplicatedMap` |
+| com.hazelcast.File | Table backed by files in a directory |
+
+**table_parameter**
+
+Parameter specific to the external storage system - connection string, serialization format etc.
```

Comment:
```suggestion
Parameter specific to the external storage system - connection string, serialization format etc.
This is an identifier, not a string. Therefore it must not be quoted as a string. It can be quoted in double
quotes if it contains special characters (such as space, period...).
```